project_role/models/project_assignment.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2018 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class ProjectAssignment(models.Model):
```

Comment:
As this is a one2many that is accessed inside project, the guideline is to put this code in `project_project.py` file instead of an isolated file.

Reply:
Ok! If possible, please refer to the relative guideline via link

Reply:
To be exact, this is not required, right?
> An exception *can be* made when the model is a model intended to be used only as a one2many model nested on the main model. In this case, you can include the model definition inside it. Example sale.order.line model *can be* together with sale.order in the file models/sale_order.py.

Reply:
Well, any of the guideline can be, but it's convenient to be homogeneous.

Reply:
In this particular case I'll surely give it a lot of consideration for some time, yet as of right now I feel it's better to opt in "not using that exception".

Reply:
OK, I won't block for this.